### PR TITLE
Update comtypes to version 1.4.6

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
None

### Summary of the issue:
There have been many updates to Comtypes since last time we updated.

### Description of user facing changes
None

### Description of development approach
* Updated comtypes to 1.4.6.
* Updated the UIAutomation comtypes wrapper.

### Testing strategy:
Should be tested during daily use.

### Known issues with pull request:
None known

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Updated the Comtypes library to version 1.4.6, enhancing functionality and performance.
	- Introduced new enumerations in UI automation to improve type safety and code clarity.
  
- **Bug Fixes**
	- Improved import structure in UIAutomationClient.py for better reliability.

- **Documentation**
	- Added notes on breaking changes in the Add-on API, advising developers to re-test and update their Add-ons.

### CODE REVIEW

**Comment by LeonarddeR** [comment]
@coderabbitai you have erroneously detected the change in changes.md as an add-on api breaking change

**Comment by LeonarddeR** [comment]
@coderabbitai you'wrote:

> Added notes on breaking changes in the Add-on API, advising developers to re-test and update their Add-ons.

You're erroneously suggesting that this info is part of this pr.

**Comment by seanbudd** [comment]
How are these files generated / where are you adding them from? can you please add this to the dev approach and/or docs somewhere

**Comment by LeonarddeR** [comment]
Any suggestions where I should add this?

**Comment by seanbudd** [comment]
@LeonarddeR either in projectDocs\dev\createDevEnvironment.md or in a separate doc referenced by this

**Comment by seanbudd** [comment]
Actually, probably `source\comInterfaces\readme.md` but please add a reference to is from projectDocs\dev\createDevEnvironment.md

**Comment by LeonarddeR** [comment]
That file reads:

> Only UIAutomation.py is not generated, UIA has historically been updated regularly and the version on
Appveyor build servers could not be guaranteed.

I wonder whether we should reconsider this position for two reasons:

1. It has been quite some time since this wrapper needed an update
2. I'm pretty sure Server 2022 has the right version on board.

**Comment by LeonarddeR** [comment]
Note that you suggested this in https://github.com/nvaccess/nvda/pull/15522#issuecomment-1738246291, but then @lukaszgo1 pointed out in https://github.com/nvaccess/nvda/pull/15522#pullrequestreview-1649777194 that Server 2019 contained an outdated version. However, appveyor is on server 2022 now.

**Comment by seanbudd** [comment]
Can you please fix up the git excludes and ruff format?

https://results.pre-commit.ci/run/github/45885732/1724844260.36kQ5pFwQpmFQXxO-uMqNw

File: source/comInterfaces_sconscript [hunk 1]
```diff
@@ -96,11 +96,9 @@ COM_INTERFACES = {
...
 	"SpeechLib.py": ("{C866CA3A-32F7-11D2-9602-00C04F8EE628}", 5, 0),
 	"AcrobatAccessLib.py": "typelibs/AcrobatAccess.tlb",
-	# We don't generate UIAutomationClient (and _944DE083_8FB8_45CF_BCB7_C477ACB2F897_0_1_0.py)
-	# because we don't know what version is available on Appveyor.
-	# Instead we locally generate the file and include it in the repository.
-	# Ensuring that adjustments performed by 'makeIDEFriendly' are incorporated.
-	# "UIAutomationClient.py": ('{944DE083-8FB8-45CF-BCB7-C477ACB2F897}', 1, 0),
+	# As long as Appveyor is on a Windows version that has an acceptable version of UIAutomationClient,
+	# we generate it at build time.
+	"UIAutomationClient.py": ('{944DE083-8FB8-45CF-BCB7-C477ACB2F897}', 1, 0),
```
Comment (seanbudd) [inline source/comInterfaces_sconscript:101]: This is failing ruff format on AppVeyor

```suggestion
	"UIAutomationClient.py": ("{944DE083-8FB8-45CF-BCB7-C477ACB2F897}", 1, 0),
```

**Review by seanbudd — APPROVED** [review]

**Pull request merged**